File: SEP_T3/src/main/java/sep/java/sep_t3/contracts/IMessageDao.java
package sep.java.sep_t3.contracts;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import sep.java.sep_t3.model.Message;

import java.util.List;

public interface IMessageDao {
    public ResponseEntity<Message> AddMessage(String messageAsJson) throws JsonProcessingException;
    public ResponseEntity<Message> GetMessage(String MID);
    public ResponseEntity<List<Message>> GetAllMessage();
    public ResponseEntity<HttpStatus> DeleteMessage(String MID);
    public ResponseEntity<Message> UpdateMessage(String messageAsJson) throws JsonProcessingException;
}

File: SEP3_T2/Contracts/DAO/IMessageDao.cs
using Entities.Model;

namespace Contracts.DAO; 

public interface IMessageDao {
    public Task<Message> AddMessage(Message message);
    public Task<Message> GetMessage(Guid MID);
    public Task<ICollection<Message>> GetAllMessage();
    public Task DeleteMessage(Guid MID);
    public Task<Message> UpdateMessage(Message message);
}